bottom/client.py

```diff
@@ -1,14 +1,20 @@
 import asyncio
 import collections
 import functools
+from typing import Any, Callable, DefaultDict, List, Optional  # noqa
```

Comment:
`DefaultDict` was added after 3.5.0, so bottom won't be strictly 3.5+.  However, upgrading patch versions and even 3.4 -> 3.5, 3.5 -> 3.6 have been relatively painless (though adoption lags, see RTD build environments).

How important is this over just using `Dict`?  Will the changes for `functools.partial` and `AbstractEventLoop` require a new python release, or will they use existing types?